archive.go

```
@@ -11,11 +12,40 @@ import (
+
+var (
+	zipMagic  = []byte{0x50, 0x4B, 0x03, 0x04}
+	gZipMagic = []byte{0x1F, 0x8B}
+)
+
+func sniffArchiveKind(path string) (ArchiveKind, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return NotArchive, fmt.Errorf("open: %w", err)
+	}
+	defer f.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `f.Close` is not checked

(errcheck)